activesupport/lib/active_support/concern.rb

```diff
@@ -104,6 +104,13 @@ def initialize
       end
     end
 
+    class UnreachableIncludedBlock < StandardError #:nodoc:
+      def initialize
+        super "This 'included' block will never be run because the Concern was already included. " +
+          "Check your code for circular references from the Concern to the including class."
```

Comment:
@dv I think @sgrif wants the actual concern name (via ` #{concern_name}`) to be mentioned in the error message (rightly so). Your current message is ok but leads to an unnecessary stack trace hunt.